### PR TITLE
added users endpoint and authentication middleware

### PR DESCRIPTION
**Pull request by gon-escalante**
## Summary

added get users endpoint and authentication middleware

## Trello Card

https://trello.com/c/DeGYHnog/4-lista-de-usuarios

### CODE REVIEW

File: app/controllers/user.js [hunk 1]
```diff
@@ -34,13 +40,21 @@ exports.signIn = (req, res, next) =>
       }
       return compareEncryptedData(req.body.password, user.password).then(isValid => {
         if (isValid) {
-          const token = getUserSessionToken({ userId: user.id });
+          const token = getUserSessionToken({ userId: user.id, admin: true });
```
Comment (picollomartin) [inline app/controllers/user.js]: You need this `admin: true` here ? All users are admin ?

File: app/services/user.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+const { user: User } = require('../models'),
+  { databaseError } = require('../errors'),
+  { DEFAULT_PAGE_LIMIT } = require('../constants');
 
 exports.createUser = ({ email, ...otherUserData }) =>
   User.findOrCreate({ where: { email }, defaults: otherUserData });
 
 exports.getUserByEmail = email => User.findOne({ where: { email } });
+
+exports.getUsers = ({ page }, next) =>
```
Comment (picollomartin) [inline app/services/user.js]: `next` in services 👀

File: app/services/user.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+  User.findAll({ limit: DEFAULT_PAGE_LIMIT, offset: page ? (page - 1) * DEFAULT_PAGE_LIMIT : 0 }).catch(err =>
+    next(databaseError(err))
+  );
+exports.getUserByEmail = email => User.findOne({ where: { email } });
```
Comment (picollomartin) [inline app/services/user.js]: Catch database error

File: app/middlewares/user.js [hunk 4]
```diff
@@ -8,3 +9,14 @@ exports.validateUserSignUpData = (req, res, next) => {
   }
   return next();
 };
+
+exports.authenticate = (req, res, next) => {
+  const { token } = req.query;
+  verifyToken(token, (err, payload) => {
+    if (payload && payload.admin) {
```
Comment (picollomartin) [inline app/middlewares/user.js]: All users should be admin for this authentication ?

**Review by picollomartin — DISMISSED** [review]

File: app/controllers/user.js (hunk 1 above)
Comment (gon-escalante) [reply app/controllers/user.js]: yes this was lazy from me, i changed this in the next pr tho :harold:

File: app/middlewares/user.js (hunk 4 above)
Comment (gon-escalante) [reply app/middlewares/user.js]: this is changed in the next pr too 🌈

**Review by igna92ts — APPROVED** [review]

**Pull request merged**